src/com/google/appengine/demos/shardedcounter/java/v2/ShardedCounter.java

```diff
@@ -234,6 +242,13 @@ private void incrementPropertyTx(final Key key, final String prop,
             LOG.log(Level.WARNING,
                     "You may need more shards. Consider adding more shards.");
             LOG.log(Level.WARNING, e.toString(), e);
+            
+            if(getShardCount() <= ATMOST_SHARDS){
+                LOG.log(Level.INFO,
+                       "Doubling Shade, for this process");
+                addShards(getShardCount()); //double the shade by adding the number of shade again
```

Comment:
Nice idea -- if you are going to call something ATMOST_SHARDS, then you should actually limit the total.  Of course, you could rename things.  Some kind of MAX shards calculation would be good.